applications/ultrasound_postprocessing/ultra_post/filters/anisotropic.py

```
@@ -0,0 +1,60 @@
+
+import cupy as cp
+
+
+def anisotropic_diffusion(
+    image: cp.ndarray,
+    num_iter: int = 10,
+    kappa: float = 0.1,
+    gamma: float = 0.1,
+    function: str = "exponential",
+) -> cp.ndarray:
+    """Perona–Malik anisotropic diffusion for edge-preserving smoothing."""
```

Ruff (0.14.10)

Docstring contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)